src/deterministic_vi/ConstraintTransforms.jl

```diff
@@ -359,22 +360,26 @@ end
 using ForwardDiff
 using ForwardDiff: Dual, jacobian!, JacobianConfig
 
+const NUM_ELBO_BOUND_PARAMS = length(CanonicalParams)
+const NUM_ELBO_FREE_PARAMS = (c = ConstraintBatch(); free_length(c.boxes[1]) + free_length(c.simplexes[1]))
+const NUM_ELBO_JACOBIAN_PARAMS = NUM_ELBO_BOUND_PARAMS * NUM_ELBO_FREE_PARAMS
+
 @compat const TransformJacobianConfig{M,N,T} = JacobianConfig{N,T,Tuple{Array{Dual{N,T},M},Vector{Dual{N,T}}}}
 
 immutable TransformDerivatives{N,T}
-    jacobian::Matrix{T}
+    jacobian::MMatrix{NUM_ELBO_BOUND_PARAMS,NUM_ELBO_FREE_PARAMS,Float64,NUM_ELBO_JACOBIAN_PARAMS}
```

Comment:
SizedMatrix may have better optimization properties, because the compiler knows about arrays.